synapse/handlers/auth.py

```diff
@@ -988,8 +990,12 @@ def complete_sso_login(
             client_redirect_url, "loginToken", login_token
         )
 
-        # if the client is whitelisted, we can redirect straight to it
-        if client_redirect_url.startswith(self._whitelisted_sso_clients):
+        # Skip the confirmation step and redirect to the client if it is whitelisted or
+        # if the server's configuration explicitly disables that step.
+        if (
+            client_redirect_url.startswith(self._whitelisted_sso_clients)
+            or not self._sso_enable_redirect_confirm
+        ):
```

Comment:
(Tiniest of nits: normally I'd put the flag check first, it makes more sense to me and allows us to short circuit the str comparison.)